geomstats/information_geometry/binomial.py

```diff
@@ -181,18 +175,240 @@ def squared_dist(self, point_a, point_b, **kwargs):
 
         Parameters
         ----------
-        point_a : array-like, shape=[...,]
+        point_a : array-like, shape=[...,1]
             Point representing a binomial distribution (probability of success).
-        point_b : array-like, shape=[...,] (same shape as point_a)
+        point_b : array-like, shape=[...,1] (same shape as point_a)
             Point representing a binomial distribution (probability of success).
 
         Returns
         -------
         squared_dist : array-like, shape=[...,]
             Squared distance between points point_a and point_b.
         """
-        return (
+        return gs.squeeze(
             4
             * self.n_draws
             * (gs.arcsin(gs.sqrt(point_a)) - gs.arcsin(gs.sqrt(point_b))) ** 2
         )
+
+    def metric_matrix(self, base_point=None):
+        """Compute the metric matrix at the tangent space at base_point.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., 1]
+            Point representing a binomial distribution.
+
+        Returns
+        -------
+        mat : array-like, shape=[..., 1, 1]
+            Metric matrix.
+        """
+        if base_point is None:
+            raise ValueError(
+                "A base point must be given to compute the " "metric matrix"
+            )
+        return gs.expand_dims(self.n_draws / (base_point * (1 - base_point)), axis=-1)
+
+    def _geodesic_ivp(self, initial_point, initial_tangent_vec):
+        """Solve geodesic initial value problem.
+
+        Compute the parameterized function for the geodesic starting at
+        initial_point with initial velocity given by initial_tangent_vec.
+
+        Parameters
+        ----------
+        initial_point : array-like, shape=[..., 1]
+            Initial point.
+
+        initial_tangent_vec : array-like, shape=[..., 1]
+            Tangent vector at initial point.
+
+        Returns
+        -------
+        path : function
+            Parameterized function for the geodesic curve starting at
+            initial_point with velocity initial_tangent_vec.
+        """
+        n_initial_points = initial_point.shape[0]
+        n_initial_tangent_vecs = initial_tangent_vec.shape[0]
+
+        if n_initial_points > n_initial_tangent_vecs:
+            raise ValueError(
+                "There cannot be more initial points than " "initial tangent vectors."
+            )
+        if n_initial_tangent_vecs > n_initial_points > 1:
+            raise ValueError(
+                "For several initial tangent vectors, "
+                "specify either one or the same number of "
+                "initial points."
+            )
+
+        c = gs.arcsin(gs.sqrt(initial_point))
+        K = initial_tangent_vec / (
```

Comment:
We try to avoid single letters for variable names. Maybe change to `constant_a` and `constant_b` or `intercept` and `slope` or anything else that seems relevant.